wrapper_writer/parsers.py

```diff
@@ -76,123 +77,84 @@ def delete_config(self):
             print("The config file does not exists")
 
 
-class ScalaParse(Parser):
+class ScalaParser(Parser):
     """
     This ScalaParse class parses a scala file, extracts the method elements and writes them out to a config file
     :param filename: The name of the file to parse
     :param config_name: The name of the yaml config file
     :param append_config: boolean value, True will overwrite an existing file, False will append to file
     """
 
-    doc_strings = []  #: This is a list of all the doc_strings gathered from the regex
+    #: The regex string specific to scala in order to get the docs, params, returns, type and name.
+    regex_string = r"/\*\*([\s,\w,\*,@,\+\.,='\[\]\-/%]*)\*/|(\w+ def|def)\s(\w+)\(([\w+:\s,\[\]=\"\(\)]*)\):\s([\w\[\]]*)"
 
-    def find_method_regex(self, retrieve_data):
+    def regex_parser(self, data):
         """
-        This function will find the raw method signature from file to be parsed
-        :return: iterable object with all methods found
+        This function takes in data as a string, then uses the regex string variable within this class. To find the
+        doc strings, name, parameters, types and return type for each function within the data.
+        It then stores this information as a Method class and saves all the methods as a list.
+        :param data:
+        :return: methods
         """
-        pattern = re.compile("def (\w+)\((.*)\): (\w+)", re.MULTILINE)
-        try:
-            pattern2 = pattern.finditer(retrieve_data)
-        except:
-            print("Nothing In There")
-        return pattern2
-
-    def find_doc_string(self, data):
-        """
-        This function will check the data for the relevant regex string, in order to pick out the doc strings from a
-        function.
-
-        :param data: The data given as a string for the regex to read.
-        """
-        matches = re.finditer(r"/\*\*([\s,\w,\*,@,\+\.,='\[\]\-/%]*)\*/", data, re.MULTILINE)
+        matches = re.finditer(self.regex_string, data, re.MULTILINE)
+        methods = []
+        doc_string = ""
         for match in matches:
-            group = match.group(1)
-            if group:
-                doc = group.replace("*", "").replace("\n     @", "\n@").replace("\n    ", "").replace("\n     ", " ")
+            if match.group(2) is None:
```

Comment:
Comments through here might be useful, its difficult to tell whats going on

Reply:
done